inductiva/scenarios.py

```diff
@@ -2,48 +2,68 @@
 
 from abc import ABC
 import tempfile
-from typing import Optional
+from typing import Optional, List
 
 from inductiva.types import Path
-from inductiva.simulation import Simulator
+from inductiva.simulation import Simulator, Command
 from inductiva.utils.files import resolve_path, get_timestamped_path
 from inductiva.utils.misc import split_camel_case
 
 
 class Scenario(ABC):
     """Base class for scenarios."""
 
-    def simulate(
-        self,
-        simulator: Simulator,
-        output_dir: Optional[Path] = None,
-        **kwargs,
-    ):
-        """Simulates the scenario."""
+    def setup_output_dir(self, output_dir: str):
+        """Setup the scenario output directory."""
         if output_dir is None:
             scenario_name_splitted = split_camel_case(self.__class__.__name__)
             output_dir_prefix = "-".join(scenario_name_splitted).lower()
             output_dir = get_timestamped_path(f"{output_dir_prefix}-output")
-
         output_dir = resolve_path(output_dir)
+        return output_dir
 
-        with tempfile.TemporaryDirectory() as input_dir:
-
-            self.gen_aux_files(simulator, input_dir)
-            self.gen_config(simulator, input_dir)
+    def setup_config(self, simulator: Simulator, input_dir: Path):
+        """Setup the scenario configuration files and arguments."""
+        self.gen_aux_files(simulator, input_dir)
+        self.gen_config(simulator, input_dir)
 
-            args = ()
-            config_filename = self.get_config_filename(simulator)
-            if config_filename:
-                args += (config_filename,)
+        args = ()
+        config_filename = self.get_config_filename(simulator)
+        if config_filename:
+            args += (config_filename,)
+        return args
 
+    def simulate(
+        self,
+        simulator: Simulator,
+        output_dir: Optional[Path] = None,
+        **kwargs,
+    ):
+        """Simulates the scenario for a single simulator call."""
+        output_dir = self.setup_output_dir(output_dir)
+        with tempfile.TemporaryDirectory() as input_dir:
+            args = self.setup_config(simulator, input_dir)
             output_path = simulator.run(
                 input_dir,
                 *args,
                 output_dir=output_dir,
                 **kwargs,
             )
+        return output_path
 
+    def simulate_pipeline(
+        self,
+        simulator: Simulator,
+        pipeline: List[Command],
```

Comment:
I don't understand this part... The Scenario is something high-level, that should not request from the user any low-level commands from a simulator... But here, the "pipeline" argument is a list of commands.... I think the pipeline of commands needs to be associated with the class simulator... and then a specific scenario (that is aware of which low-level commands need to be run) will tell the simulator to run that pipeline of commands, but the base class "Scenario" wouldn't expose that in a method (i.e. there should not be a simulate_pipeline in Scenario)... No?

Reply:
@hpenedones I agree, but I was having difficulties because I wanted to keep the other existing scenarios working and didnt want to change the input arguments to the simulate command. I've altered the code. Is this more inline to what you had in mind?